src/file_storage.cpp

```diff
@@ -564,7 +564,7 @@ namespace libtorrent
 		else
 		{
 			if (m_files.empty())
-				m_name = split_path(path).c_str();
+				m_name = split_path(path);
```

Comment:
I think this is the root of the problems in the CI, since internally there is a `ret.append(1, '\0');` that seems to make a difference in the internal char*.